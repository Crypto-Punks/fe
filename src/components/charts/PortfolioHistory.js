import { Line } from 'react-chartjs-2';
import React, { useEffect, useState } from 'react';
import moment from 'moment';
import { getTotals } from '../../services/total';
import { getTrades } from '../../services/trades';

const PortfolioHistory = () => {
  const [intervals, setIntervals] = useState([]);
  const [totals, setTotals] = useState([]);

  const PortfolioHistoryOptions = {
    title: {
      display: true,
      text: 'Portfolio History'
    },
    animation: {
      duration: 1000,
      easing: 'linear',
      showlines: true
    }
  };

  useEffect(() => {
    Promise.all([
      getTotals(),
      getTrades()
    ])
      .then(([totals]) => {
        setIntervals(totals.map(item =>  moment(item.timestamp).format('MMM Do YYYY')));
        setTotals(totals.map(item => item.totals.reduce((acc, val) => {
          acc += val.value;
          return acc;
        }, 0)));
      });
  }, []);

  return (
    <Line data={{ 
      labels: intervals, 
      datasets: [{ 
        label: 'net worth', 
        data: totals, 
        backgroundColor: 'rgba(255, 0, 0, 0.2)',
        borderWidth: .5,
        borderColor: 'rgba(0, 0, 0, 0.2)'
      }] 
    }} 
    options={PortfolioHistoryOptions}
    /> 
  );
};

export default PortfolioHistory;
