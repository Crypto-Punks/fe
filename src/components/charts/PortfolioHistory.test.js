import React from 'react';
import { shallow } from 'enzyme';
import PortfolioHistory from './PortfolioHistory';

describe('PortfolioHistory component', () => {
  it('renders PortfolioHistory', () => {
    const wrapper = shallow(<PortfolioHistory />);
    expect(wrapper).toMatchSnapshot();
  });
});
