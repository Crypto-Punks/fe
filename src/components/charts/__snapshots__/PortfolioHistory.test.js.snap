// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioHistory component renders PortfolioHistory 1`] = `
<Line
  data={
    Object {
      "datasets": Array [
        Object {
          "backgroundColor": "rgba(255, 0, 0, 0.2)",
          "borderColor": "rgba(0, 0, 0, 0.2)",
          "borderWidth": 0.5,
          "data": Array [],
          "label": "net worth",
        },
      ],
      "labels": Array [],
    }
  }
  options={
    Object {
      "animation": Object {
        "duration": 1000,
        "easing": "linear",
        "showlines": true,
      },
      "title": Object {
        "display": true,
        "text": "Portfolio History",
      },
    }
  }
/>
`;
